### PR TITLE
cas api: add casadm commands for failover

### PR DESCRIPTION
**Pull request by karolinavelkaja**
Signed-off-by: Karolina Rogowska <karolina.rogowska@intel.com>

### CODE REVIEW

File: test/functional/api/cas/casadm.py [hunk 1]
```diff
@@ -42,6 +42,35 @@ def start_cache(cache_dev: Device, cache_mode: CacheMode = None,
...
+    _cache_id = None if cache_id is None else str(cache_id)
+    output = TestRun.executor.run(start_standby_cmd(
+        cache_dev=cache_dev.path, cache_line_size=_cache_line_size, cache_id=_cache_id,
+        shortcut=shortcut))
+    if output.exit_code != 0:
+        raise CmdException("Failed to start standby cache.", output)
+    return Cache(cache_dev)
+
+
+def detach_cache(cache_id: int, shortcut: bool = False):
```
[the remark this answers is not in the document]
Comment (Deixx) [reply test/functional/api/cas/casadm.py:58]: Maybe you're right. But I'd leave at least `start_standby_cache` as is - it is after all a start-cache command with a flag.

**Review by Deixx — CHANGES_REQUESTED** [review]

File: test/functional/api/cas/casadm.py [hunk 2]
```diff
@@ -42,6 +42,36 @@ def start_cache(cache_dev: Device, cache_mode: CacheMode = None,
...
 
 
+def start_standby_cache(cache_dev: Device, cache_line_size: CacheLineSize,
+                        cache_id: int, shortcut: bool = False,
+                        kernel_params: KernelParameters = KernelParameters()):
+    if kernel_params != KernelParameters.read_current_settings():
+        reload_kernel_module("cas_cache", kernel_params.get_parameter_dictionary())
+
+    output = TestRun.executor.run(start_cmd(
+        cache_dev=cache_dev.path, cache_line_size=cache_line_size, cache_id=cache_id,
```
Comment (Deixx) [inline test/functional/api/cas/casadm.py:52]: we should pass strings as arguments to start_cmd, that is we have to "stringify" cache_line_size and cache_id

File: test/functional/api/cas/cli.py [hunk 3]
```diff
@@ -92,19 +92,35 @@ def flush_core_cmd(cache_id: str, core_id: str, shortcut: bool = False):
...
         command += (" -x " if shortcut else " --cache-line-size ") + cache_line_size
     if cache_id is not None:
         command += (" -i " if shortcut else " --cache-id ") + cache_id
-    if force:
-        command += " -f" if shortcut else " --force"
-    if load:
-        command += " -l" if shortcut else " --load"
+    if standby:
+        command += " --failover-standby "
+    else:
```
Comment (Deixx) [inline test/functional/api/cas/cli.py:104]: Remove 'else' - methods in cli.py are meant to build a command the same way a user can (even with incorrect configuration) and in general they should not be called directly (excluding some cases like negative tests) but through casadm API which takes care of proper arguments being passed down.

**Review by Deixx — CHANGES_REQUESTED** [review]

**Review by pdebski21 — APPROVED** [review]

**Comment by karolinavelkaja** [comment]
This PR needs to be updated to new design / function names when they will be present on master branch.

**Comment by pdebski21** [comment]
new design naming changes and review suggestions are present in new PR:
https://github.com/Open-CAS/open-cas-linux/pull/1075

**Pull request closed without merging**